pkg/hintrunner/zero/zerohint_test.go

```diff
@@ -0,0 +1,385 @@
+				},
+				makeHinter: func(ctx *testContext) hinter.Hinter {
+					return newIsNNOutOfRangeHint(ctx.operanders["a"])
+				},
+				check: apValueEquals(feltUint64(0)),
+			},
+
+			{
+				operanders: []*operander{
+					{name: "a", kind: apRelative, felt: feltInt64(1)},
+				},
+				makeHinter: func(ctx *testContext) hinter.Hinter {
+					return newIsNNOutOfRangeHint(ctx.operanders["a"])
+				},
+				check: apValueEquals(feltUint64(0)),
+			},
+
+			{
+				operanders: []*operander{
+					{name: "a", kind: fpRelative, felt: feltInt64(-1)},
+				},
+				makeHinter: func(ctx *testContext) hinter.Hinter {
+					return newIsNNOutOfRangeHint(ctx.operanders["a"])
+				},
+				check: apValueEquals(feltUint64(1)),
+			},
+
+			{
+				operanders: []*operander{
+					{name: "a", kind: apRelative, felt: &utils.FeltMax128},
+				},
+				makeHinter: func(ctx *testContext) hinter.Hinter {
+					return newIsNNOutOfRangeHint(ctx.operanders["a"])
+				},
+				check: apValueEquals(feltUint64(1)),
+			},
+
+			{
+				operanders: []*operander{
+					{name: "a", kind: apRelative, felt: feltAdd(&utils.FeltMax128, feltInt64(1))},
+				},
+				makeHinter: func(ctx *testContext) hinter.Hinter {
+					return newIsNNOutOfRangeHint(ctx.operanders["a"])
+				},
+				check: apValueEquals(feltUint64(1)),
+			},
+
+			{
+				operanders: []*operander{
+					{name: "a", kind: apRelative, felt: feltAdd(&utils.FeltMax128, feltInt64(-1))},
+				},
+				makeHinter: func(ctx *testContext) hinter.Hinter {
+					return newIsNNOutOfRangeHint(ctx.operanders["a"])
+				},
+				check: apValueEquals(feltUint64(0)),
+			},
+		},
+	}
+
+	runTest := func(t *testing.T, tc testCase) {
+		vm := VM.DefaultVirtualMachine()
+		ctx := &hinter.HintRunnerContext{}
+		if tc.vminit != nil {
+			tc.vminit(vm)
+		}
+
+		testCtx := &testContext{
+			vm:          vm,
+			hintContext: ctx,
+			operanders:  make(map[string]hinter.ResOperander),
+		}
+
+		// There are always a few extra values on the memory stack
+		// above FP to make AP-based and FP-based addressing makes more sense.
+		// These elements are identical to their index: mem[0] is 0, mem[1] is 1.
+		//
+		// Since these values are *below* FP, they can be considered to be arguments
+		// to the function; accessed as [fp-1], etc.
+		extraValues := []*fp.Element{
+			feltUint64(0),
+			feltUint64(1),
+			feltUint64(2),
+			feltUint64(3),
+		}
+		for _, v := range extraValues {
+			runnerutil.WriteTo(vm, VM.ExecutionSegment, vm.Context.Ap, memory.MemoryValueFromFieldElement(v))
+			vm.Context.Ap++
+		}
+		// FP points *after* the last extra value.
+		vm.Context.Fp = uint64(len(extraValues))
+
+		for _, o := range tc.operanders {
+			switch o.kind {
+			case apRelative, fpRelative:
+				o.memoryOffset = vm.Context.Ap
+				runnerutil.WriteTo(vm, VM.ExecutionSegment, vm.Context.Ap, memory.MemoryValueFromFieldElement(o.felt))
+				vm.Context.Ap++
+
+			case immediate:
+				// Nothing to do.
+
+			default:
```

Comment:
It appears you are not including DoubleDeref or BinaryOperation types, could you explain the issue with those?

Reply:
I wasn't sure whether to go on with that or not.
It's good if we want to test the addressing modes along the way, but perhaps it's easier to have these tests to be separated.

Or we can select the addressing mode from a table with all combinations for every test, just to increase the harness. This would remove the operander kind from the test case description object.

Reply:
These operander types can be important for ones where you need to get their address. It won't work with immediate, etc.

Reply:
I don't think trying all the combinations in every test is the best idea. For these tests is ok to only use the current types (as it is), I actually haven't dealt with any different types in my examples yet. 
I was only wondering if there was any other reason not to add them here. Let's leave it like this and have separate ones to check we are getting the address correctly.